commit-status-publisher-server/src/main/java/jetbrains/buildServer/commitPublisher/CommitStatusPublisherListener.java

```diff
@@ -172,47 +177,85 @@ public boolean run(@NotNull CommitStatusPublisher publisher, @NotNull BuildRevis
   private boolean isPublishingDisabled(SBuildType buildType) {
     String publishingEnabledParam = buildType.getParameterValue(PUBLISHING_ENABLED_PROPERTY_NAME);
     return "false".equals(publishingEnabledParam)
-           || !(TeamCityProperties.getBooleanOrTrue(PUBLISHING_ENABLED_PROPERTY_NAME)
-                || "true".equals(publishingEnabledParam));
+      || !(TeamCityProperties.getBooleanOrTrue(PUBLISHING_ENABLED_PROPERTY_NAME)
+      || "true".equals(publishingEnabledParam));
+  }
+
+  private boolean isPublishingToDependenciesEnabled(SBuildType buildType) {
+    String publishingEnabledParam = buildType.getParameterValue(PUBLISHING_TO_DEPENDENCIES_ENABLED_PROPERTY_NAME);
+    return StringUtil.areEqualIgnoringCase("true", publishingEnabledParam);
   }
 
   private void logStatusNotPublished(@NotNull Event event, @NotNull String buildDescription, @NotNull CommitStatusPublisher publisher, @NotNull String message) {
     LOG.info(String.format("Event: %s, build %s, publisher %s: %s", event.getName(), buildDescription, publisher.toString(), message));
   }
 
   private void runForEveryPublisher(@NotNull Event event, @NotNull SBuildType buildType, @NotNull SBuild build, @NotNull PublishTask task) {
+    final String description = LogUtil.describe(build);
+    Set<BuildRevision> publishedRevisions = new HashSet<BuildRevision>();
+    runForEveryPublisher(event, buildType, build, task, description, publishedRevisions);
+
+    if (!isPublishingToDependenciesEnabled(buildType)) {
+      return;
+    }
+
+    for (BuildPromotion bp : build.getBuildPromotion().getAllDependencies()) {
+      runForEveryPublisher(event, bp.getBuildType(), bp.getAssociatedBuild(), task, description, publishedRevisions);
```

Comment:
bp.getAssociatedBuild() may in theory return a null value, which is not accepted by runForEveryPublisher method.